core/file_walker.go

```diff
@@ -59,6 +55,7 @@ func getSubTreeConcurrently(path string, parent *File, readDir ReadDir, ignoredF
 	}
 	dirName, name := filepath.Split(path)
 	result.Files = make([]*File, 0, len(entries))
+	var mutex sync.Mutex
```

Comment:
Mutex is only guarding one resource. That means that we are not blocking goroutines on unrelated folders